plugins/bios/fu-plugin-bios.c

```diff
@@ -79,22 +43,22 @@ fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 #if defined(__x86_64__) || defined(__i386__)
 	g_autoptr(GError) error_local = NULL;
 	if (!fu_efivar_supported (&error_local)) {
-		const gchar *reason = "Firmware can not be updated in legacy BIOS mode, switch to UEFI mode";
-		g_warning ("%s", error_local->message);
-		return fu_plugin_bios_create_dummy (plugin, reason, error);
+		fu_plugin_add_flag (plugin, FWUPD_PLUGIN_FLAG_LEGACY_BIOS);
+		fu_plugin_add_flag (plugin, FWUPD_PLUGIN_FLAG_USER_WARNING);
+		return TRUE;
 	}
 #endif
 
 	/* get the directory of ESRT entries */
 	sysfsfwdir = fu_common_get_path (FU_PATH_KIND_SYSFSDIR_FW);
 	esrt_path = g_build_filename (sysfsfwdir, "efi", "esrt", NULL);
 	if (!g_file_test (esrt_path, G_FILE_TEST_IS_DIR)) {
-		const gchar *reason = "UEFI Capsule updates not available or enabled";
-		return fu_plugin_bios_create_dummy (plugin, reason, error);
+		fu_plugin_add_flag (plugin, FWUPD_PLUGIN_FLAG_CAPSULES_UNSUPPORTED);
+		fu_plugin_add_flag (plugin, FWUPD_PLUGIN_FLAG_USER_WARNING);
+		return TRUE;
 	}
 
 	/* we appear to have UEFI capsule updates */
-	fu_plugin_set_enabled (plugin, FALSE);
-
+	fu_plugin_add_flag (plugin, FWUPD_PLUGIN_FLAG_DISABLED);
```

Comment:
this feels like it shouldn't just be a flag should it?

Reply:
It's just a tiny boolean attribute about the plugin, it seemed sane to fold in a 32bit boolean :( into the 64bit flag.